### ISSUE TITLE
Rule Request: Prefer Nimble matchers

### ISSUE THREAD

**Issue by marcelofabri** [realm/SwiftLint#3293]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### New rule request

We use [Nimble](https://github.com/Quick/Nimble) in our unit tests. For consistency, we want only Nimble matchers to be used, instead of XCTest ones. Right now we're using this regex rule:

```yml
  xctassert:
    regex: "XCTAssert\\w*"
    name: "Use Nimble"
    match_kinds:
      - identifier
    message: "Use Nimble matchers instead of XCTest directly"
    severity: warning
```

1. Why should this rule be added? Share links to existing discussion about what
   the community thinks about this.

Nimble is a popular matcher library for testing.

2. Provide several examples of what _would_ and _wouldn't_ trigger violations.

```swift
// ok
expect(value) == 1
expect(value).to(beNil())

// not ok
XCTAssertEqual(value, 1)
XCTAssertNil(value)
```

3. Should the rule be configurable, if so what parameters should be configurable?

Just severity for now

4. Should the rule be opt-in or enabled by default? Why?
   See [README.md](../README.md#opt-in-rules) for guidelines on when to mark a rule as opt-in.

Opt-in for sure, as not all projects use Nimble
